validator/client/validator.go

```diff
@@ -238,8 +247,33 @@ func (v *validator) WaitForActivation(ctx context.Context) error {
 
 	validatingKeys, err := v.keyManager.FetchValidatingPublicKeys(ctx)
 	if err != nil {
-		return errors.Wrap(err, "could not fetch validating keys")
+		return errors.Wrap(err, msgCouldNotFetchKeys)
 	}
+	if len(validatingKeys) == 0 {
+		log.Warn(msgNoKeysFetched)
+
+		ticker := time.NewTicker(keyRefetchPeriod)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ticker.C:
+				keys, err := v.keyManager.FetchValidatingPublicKeys(ctx)
+				if err != nil {
+					return errors.Wrap(err, msgCouldNotFetchKeys)
+				}
+				if len(keys) == 0 {
+					log.Warn(msgNoKeysFetched)
+					continue
+				}
+				validatingKeys = keys
```

Comment:
Should this return after or do we need to continuously fetch validating public keys?

Reply:
After this statement we jump out of `select` and immediately hit `break`, which will jump out of the `for` loop. I had a `break` statement here initially, but DeepSource complained that it's redundant.

Reply:
Got it. Thanks for the comment!